### PR TITLE
Improved error handling for message search method

### LINKED ISSUE #444: Missed errors in search method
Hi,
It's my first time posting there, so please forgive for any mistake made.

I've recently encountered a weird error when I was doing search with Since date condition(that returned 0 messages as a result, which is incorrect in my case):

`PHP Notice:  Unknown: SEARCH Database is not full text indexed (errflg=2) in Unknown on line 0`

It appears to be a system caused issue and was resolved by our admins.
However, I think it will be better if this library will gracefully catch and address all similar errors.

So, we could move error checking logic above like this
```

public function getMessages(ConditionInterface $search = null, int $sortCriteria = null, bool $descending = false, string $charset = null): MessageIteratorInterface
    {
        ...
       if (false !== ($error = \imap_last_error())) {
            throw new InvalidSearchCriteriaException(\sprintf('Invalid search criteria [%s] - %s', $query, $error));
       }
       if (false === $messageNumbers) {
            // imap_search can also return false
            $messageNumbers = [];
        }

        return new MessageIterator($this->resource, $messageNumbers);
    }

```
This way any unexpected error will be reported right away without producing imprecise results.

### PR DESCRIPTION
**Pull request by ikarol**
closes #444 

I wasn't able to reproduce(through tests) the issue reported in #444  as it's server specific and not trivial. However, this change doesn't break existing tests for search functionality.

### CODE REVIEW

**Comment by Slamdunk** [comment]
See https://github.com/ddeboer/imap/pull/436/files#r370508077

**Comment by ikarol** [comment]
> See https://github.com/ddeboer/imap/pull/436/files#r370508077

Hi,
there is one key difference in my use case and implementation vs #436 
In my use case  \imap_search() and \imap_sort() weren't returning false, but empty array(implying that the operation has been executed successfully), so no exception was thrown(and if you were to merge #436 , it wouldn't catch that as well for the same reason) and this error emitted notice at the end(see #444 for description) that broke my script. This was the error caused by mailbox missconfiguration.

So, my goal is to catch this kind of errors and prevent them from producing incorrect results. Now, I understand that such errors might not be relevant to InvalidSearchCriteriaException and we might need to introduce new exception type which will be responsible for errors caused "in between" search operation.

What do you think?

**Comment by Slamdunk** [comment]
Ok, if so please use `false !== \imap_last_error()` instead of `imap_errors()`: `imap_last_error` doesn't alter the error stack, so the exceptions can stay the same

**Comment by ikarol** [comment]
Do you think we need a new exception type for this to separate it from search criteria specific cases?

**Comment by Slamdunk** [comment]
It is a good idea, but not for now it's good as is

File: src/Mailbox.php [hunk 1]
```diff
@@ -181,11 +181,14 @@ public function getMessages(ConditionInterface $search = null, int $sortCriteria
             }
             $messageNumbers = \imap_search(...$params);
         }
+        $error = \imap_last_error();
+        if (false !== $error) {
+            // this way all errors occurred during search will be reported
+            throw new InvalidSearchCriteriaException(
+                \sprintf('Invalid search criteria [%s] - %s', $query, $error)
```
Comment (Slamdunk) [inline src/Mailbox.php]: There is no need for saving the `imap_last_error` output and putting it into the exception message: exceptions already output all the error stack

Comment (ikarol) [reply src/Mailbox.php]: Sorry, already fixed that in a [new commit](https://github.com/ddeboer/imap/pull/445/commits/06bffa2298be4fc88583bb7b7909ae4d32aa951e)

**Comment by Slamdunk** [comment]
Thank you!

**Pull request merged**